pennylane/transforms/decompositions/two_qubit_unitary.py

```diff
@@ -66,38 +54,45 @@ def _convert_to_su4(U):
     # Convert to SU(4) if it's not close to 1
     if not math.allclose(det, 1.0):
         exp_angle = -1j * math.cast_like(math.angle(det), 1j) / 4
-        U = math.cast_like(U, det) * qml.math.exp(exp_angle)
+        U = math.cast_like(U, det) * math.exp(exp_angle)
 
     return U
 
 
-def _select_rotation_angles(U):
-    r"""Choose the rotation angles of RZ, RY in the two-qubit decomposition.
-    They are chosen as per Proposition V.1 in quant-ph/0308033 and are based
-    on the phases of the eigenvalues of :math:`E^\dagger \gamma(U) E`, where
+def _compute_num_cnots(U):
+    r"""Compute the number of CNOTs required to implement U. This is based on
+    the trace of
```

Comment:
this is a... _very_ cool function. Are there use-cases for it beyond 2-qubit decomp? Maybe for specifications? Probably not, right, since you need to have a unitary matrix representation of the circuit, which is not likely to be the case?

Reply:
(when it says 'required to implement U', is this the optimum/minimum number of CNOTs required to implement U? Or is it specific to this decomposition?)

Reply:
I believe this is minimal as it's based on properties of the matrix / some group theory stuff related to the character of the gamma(U) function of the elements (I don't know enough about this to comment further). But in any case, you can show just through circuits and parameter counting that 3 is the maximum number needed, but there are of course special cases where you can get by with just 1 or 2. I'm really disappointed I can't figure out consistent decompositions for those because the circuits are really slick, and minimizing CNOTs is super important!